### ISSUE TITLE
QHP index contains duplicated file entries

### ISSUE THREAD

**Issue by ferdymercury** [doxygen/doxygen#9138]
**Describe the bug**
As discussed in https://github.com/doxygen/doxygen/issues/9094#issuecomment-1046799213, the doxygen-generated QHP file contains duplicated entries in the file name list. (This QHP generation is disabled by default, but one can activate it to create a QtHelp book.)

**Expected behavior**
No duplicates are found in index.qhp

**Screenshots**
```
      <file>functions.html</file>
      <file>functions.html</file>
      <file>functions.html</file>
      <file>functions.html</file>
      <file>functions.html</file>
      <file>functions_func.html</file>
      <file>functions_func.html</file>
      <file>functions_func.html</file>
      <file>functions_func.html</file>
      <file>functions_func.html</file>
      <file>files.html</file>
      <file>globals.html</file>
      <file>globals.html</file>
      <file>globals.html</file>
      <file>globals_func.html</file>
      <file>globals_func.html</file>
      <file>globals_func.html</file>
```

**To Reproduce**
See reproducer at https://github.com/doxygen/doxygen/issues/9094

**Version**
1.9.4 (15dae504a167d838873053cc96417d8beae78df7)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9139

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.4.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
